lib/pbench/agent/tool_meister.py

```diff
@@ -1316,121 +1316,120 @@ def _send_directory(self, directory, uri, ctx):
         target_dir = directory.name
         parent_dir = directory.parent
         tar_file = parent_dir / f"{target_dir}.tar.xz"
-        o_file = parent_dir / f"{target_dir}.tar.out"
-        e_file = parent_dir / f"{target_dir}.tar.err"
+        tar_args = [
+            self.tar_path,
+            "--create",
+            "--xz",
+            "--force-local",
+            f"--file={tar_file}",
+            target_dir,
+        ]
         try:
             # Invoke tar directly for efficiency.
-            with o_file.open("w") as ofp, e_file.open("w") as efp:
-                cp = subprocess.run(
-                    [
-                        self.tar_path,
-                        "--create",
-                        "--xz",
-                        "--force-local",
-                        f"--file={tar_file}",
-                        target_dir,
-                    ],
-                    cwd=parent_dir,
-                    stdin=None,
-                    stdout=ofp,
-                    stderr=efp,
-                )
+            cp = subprocess.run(
+                tar_args,
+                cwd=parent_dir,
+                stdin=None,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+            )
         except Exception:
             self.logger.exception("Failed to create tar ball '%s'", tar_file)
             failures += 1
         else:
             try:
                 if cp.returncode != 0:
-                    self.logger.error(
-                        "Failed to create tar ball; return code: %d", cp.returncode,
+                    self.logger.warning(
+                        "Failed to create tar ball first time; Error: %s, re-trying with --warning=none",
+                        str(cp.stdout),
                     )
-                    failures += 1
-                else:
+                    # record the tar error we encountered first time
+                    with open(target_dir / "tar_command_error") as file:
+                        file.write(str(cp.stdout))
+                    tar_args.insert(2, "--warning=none")
                     try:
-                        (_, tar_md5) = md5sum(tar_file)
+                        cp = subprocess.run(
+                            tar_args,
+                            cwd=parent_dir,
+                            stdin=None,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.STDOUT,
+                        )
+                        assert (
+                            cp.returncode == 0
+                        ), "Failed to create tar file second time"
                     except Exception:
```

Comment:
The assert seems unnecessarily harsh; for one thing, we won't be logging the exception and incrementing `failures` ... or ... well, actually we *will*, because your `except Exception` will immediately intercept the `assert` 😄 

Also, again, the point of an `assert` is that it sublimates without residue in a "production" `-O` build, and we'll still want to recognize and handle failure of the retry in that case.

Reply:
Yeah, the command should theoretically succeed because we specify `--warning=none` , so this assertions is to just reiterate that we got `returncode=0` with this option and if any exception occurs we still have the exception block to catch it. If the returncode is not 0 for any reason for the 2nd time even after specifying `--warning=none` we should exit isn't it? and failed assertion will do that for us.

Reply:
Look at the call sites:

```
        failures = self._send_directory(tool_dir, "tool-data", tool_data_ctx)

        if failures == 0:
            del self.directories[directory]

        self._send_client_status(
            "success" if failures == 0 else f"{failures} failures sending tool data"
        )
```

If something goes wrong with the `tar` and we `assert`, we'll never send the failure message to the TDS, and the whole thing hangs up because TDS never got all the send confirmations. That would be bad.

Reply:
Peter illuminated this for me:  his request for this PR comes from a scenario which Lukas Doktor hit where Pbench was trying to capture the run results while SAR was still writing them, which caused the files to change while they were being archived, which `tar` gets upset about.  So, the point of the retry is to overcome `tar`'s concerns on that front.

I believe that Peter recommended that we not concern ourselves with other modes of failure, here (which, perhaps, is a good idea).  That being the case, I would recommend tailoring the error handling here toward the "files changed" failure that Lukas hit, to the extent that it is possible -- that seems like a realistic and specific problem scenario, and it seems like one that we might address without "boiling the ocean"....  (And, it's something that we could explain in the code....)

Reply:
> I would recommend tailoring the error handling here toward the "files changed" failure that Lukas hit, to the extent that it is possible

So you mean instead of doing `warning=None` we do `warning=file-changed --ignore-failed-read` that should suppress the warnings about changing files while we are archiving

Reply:
We'll never be bullet-proof here, so that suggestion has some merit. If there are *particular* default warnings we know we can successfully ignore to have a better chance of getting a dataset (and aren't 100% sure we understand the consequences of ignoring *all* warnings... which I'm certainly not), then picking the ones we know may make sense. With that said, I don't object to `--warning=none` as a generic "let's just do our best to get some artifact we can bring back to the museum".

Reply:
My intention was to change to react only to specific modes of failure, but that seems to be prohibitively difficult (i.e., it requires parsing `stderr`, which seems like a bad idea).

Once we decide to retry, I think we should grab whatever we can get and flee (although, I am slightly worried that by the time we get back to the museum we will have forgotten that we left in a panic...).